CodeEdit/Features/Settings/Pages/ThemeSettings/ThemeSettingsThemeDetails.swift

```diff
@@ -11,108 +11,169 @@ struct ThemeSettingsThemeDetails: View {
+                                "Numbers",
+                                color: $theme.editor.numbers.swiftColor
+                            )
+                            Divider().padding(.horizontal, 10)
+                            ThemeSettingsThemeToken(
+                                "Strings",
+                                color: $theme.editor.strings.swiftColor
+                            )
+                            Divider().padding(.horizontal, 10)
+                            ThemeSettingsThemeToken(
+                                "Characters",
+                                color: $theme.editor.characters.swiftColor
+                            )
+                            Divider().padding(.horizontal, 10)
+                            ThemeSettingsThemeToken(
+                                "Comments",
+                                color: $theme.editor.comments.swiftColor
+                            )
+                        }
+                        .background(theme.editor.background.swiftColor)
+                        .padding(-10)
+                        .colorScheme(
+                            theme.appearance == .dark
+                            ? .dark
+                            : theme.appearance == .light
+                            ? .light : colorScheme
+                        )
+                    }
                 }
-            }.formStyle(.grouped)
+                .disabled(theme.isBundled)
+            }
+            .formStyle(.grouped)
             Divider()
             HStack {
+                if !themeModel.isAdding {
+                    Button {
+                        if let fileURL = theme.fileURL {
+                            themeModel.duplicate(fileURL)
+                        }
+                    } label: {
+                        Text("Duplicate")
+                            .frame(minWidth: 56)
+                    }
+                    if !theme.isBundled {
+                        Button(role: .destructive) {
+                            themeModel.delete(theme)
+                            dismiss()
+                        } label: {
+                            Text("Delete")
+                                .foregroundStyle(.red)
+                                .frame(minWidth: 56)
+                        }
+                    }
+                }
                 Spacer()
                 Button {
-                    theme = initialTheme
+                    if themeModel.isAdding {
+                        themeModel.delete(theme)
+                    } else {
+                        themeModel.cancelDetails(theme)
+                    }
+
                     dismiss()
                 } label: {
                     Text("Cancel")
+                        .frame(minWidth: 56)
                 }
                 .buttonStyle(.bordered)
                 Button {
+                    themeModel.rename(to: theme.displayName, theme: theme)
```

Comment:
One thing I noticed while messing around is I expected my new/saved theme to be selected after saving. Could we make a call to `ThemeSettingsView.activateTheme` here by passing it to this view like we do the rows?

Reply:
Is it good practice to call another views function? I wonder if we can move the activateTheme function inside `ThemeModel`.

Reply:
@thecoolwinter okay, I've made a change that should address this by moving activateTheme to ThemeModel.